patches/roslyn-analyzers/0001-WIP-for-source-build.patch

```diff
@@ -0,0 +1,1087 @@
++    <Exec Command='"$(DotNetExecutible)" "$(_GenerateGlobalAnalyzerConfigsPath)" "$(_GeneratedGlobalAnalyzerConfigsDir)" "$(NuspecPackageId)" "$(PackageTargetsFileDir)" "$(PackageTargetsFileName)" "@(AnalyzerRulesetAssembly)" "$(ArtifactsBinDir)" "$(Configuration)" "%(AnalyzerRulesetAssembly.TargetFramework)" "$(ReleaseTrackingOptOut)"' />
+ 
+     <ItemGroup Condition="Exists('$(PackageTargetsFileDir)\$(PackageTargetsFileName)')">
+       <AnalyzerNupkgFile Include="$(PackageTargetsFileDir)\$(PackageTargetsFileName)"/>
+@@ -132,6 +137,8 @@
+       <_CscToolPath Condition="!HasTrailingSlash('$(_CscToolPath)')">$(_CscToolPath)\</_CscToolPath>
+     </PropertyGroup>
+ 
+-    <Exec Command='"$(_CscToolPath)csi.exe" "$(RepoRoot)eng\GenerateAnalyzerNuspec.csx" "$(NuspecFile)" "$(AssetsDir)\" "$(MSBuildProjectDirectory)" "$(Configuration)" "$(TargetFrameworksForPackage)" "@(_NuspecMetadata)" "@(AnalyzerNupkgFile)" "@(AnalyzerNupkgFolder)" "@(AnalyzerNupkgAssembly)" "@(AnalyzerNupkgDependency)" "@(AnalyzerNupkgLibrary)" "$(_GeneratedRulesetsDir)" "$(_GeneratedEditorconfigsDir)" "@(AnalyzerLegacyRuleset)" "$(ArtifactsBinDir)\" "$(AnalyzerDocumentationFileDir)" "$(AnalyzerDocumentationFileName)" "$(AnalyzerSarifFileDir)" "$(AnalyzerSarifFileName)" "$(AnalyzerConfigurationFileDir)" "$(AnalyzerConfigurationFileName)" "$(_GeneratedGlobalAnalyzerConfigsDir)"' />
++    <Exec Command='"$(DotNetExecutible)" "build" "-c" "Release" "$(RepoRoot)eng\GenerateAnalyzerNuspec\GenerateAnalyzerNuspec.csproj"'
++          EnvironmentVariables="MSBUILDDISABLENODEREUSE=1;UseSharedCompilation=false;UseRazorBuildServer=false"/>
++    <Exec Command='"$(DotNetExecutible)" "$(RepoRoot)\artifacts\bin\GenerateAnalyzerNuspec\Release\net5.0\GenerateAnalyzerNuspec.dll" "$(NuspecFile)" "$(AssetsDir)" "$(MSBuildProjectDirectory)" "$(Configuration)" "$(TargetFrameworksForPackage)" "@(_NuspecMetadata)" "@(AnalyzerNupkgFile)" "@(AnalyzerNupkgFolder)" "@(AnalyzerNupkgAssembly)" "@(AnalyzerNupkgDependency)" "@(AnalyzerNupkgLibrary)" "$(_GeneratedRulesetsDir)" "$(_GeneratedEditorconfigsDir)" "@(AnalyzerLegacyRuleset)" "$(ArtifactsBinDir)" "$(AnalyzerDocumentationFileDir)" "$(AnalyzerDocumentationFileName)" "$(AnalyzerSarifFileDir)" "$(AnalyzerSarifFileName)" "$(AnalyzerConfigurationFileDir)" "$(AnalyzerConfigurationFileName)" "$(_GeneratedGlobalAnalyzerConfigsDir)"' />
+   </Target>
+-</Project>
+\ No newline at end of file
++</Project>
+diff --git a/eng/GenerateAnalyzerNuspec/GenerateAnalyzerNuspec.csproj b/eng/GenerateAnalyzerNuspec/GenerateAnalyzerNuspec.csproj
+new file mode 100644
+index 000000000..208270428
+--- /dev/null
++++ b/eng/GenerateAnalyzerNuspec/GenerateAnalyzerNuspec.csproj
+@@ -0,0 +1,8 @@
++<Project Sdk="Microsoft.NET.Sdk">
++
++  <PropertyGroup>
++    <OutputType>Exe</OutputType>
++    <TargetFramework>net5.0</TargetFramework>
++  </PropertyGroup>
```

Comment:
Is there anything I need to do to make sure I am not accidentally using prebuilts here?

Reply:
In general, even the Microsoft build wants to be able to build a basic app without downloading everything (simple "offline" scenarios) so a clean/basic proj like this should be ok. Looking at the annotated prebuilt report for the `project.assets.json` of this proj is the way to tell from the end results.